### DISCUSSION TITLE
How can I include an ArgsTable in my MDX documentation for an embedded story?

### DISCUSSION THREAD

**Discussion by vinman1** [storybookjs/storybook#16549] [Help]
Hi,

I'm using `@storybook/angular`, version: `6.4.0-beta.23` with `@angular/core`, version `12.2.11` . 

I've followed the guidance [here](https://storybook.js.org/docs/angular/writing-docs/mdx#embedding-stories) for embedding a defined story as part of my MDX documentation. I'd like to include an `ArgsTable` in my MDX for one of my embedded stories, using these [instructions](https://storybook.js.org/docs/angular/writing-docs/doc-blocks#for-stories).

My args table only seems to render correctly if I use either of `PRIMARY_STORY` or `CURRENT_SELECTION` as values for the `story` prop, e.g.:

```mdx
<ArgsTable story={PRIMARY_STORY}/>
```

Unfortunately, this associates the `ArgsTable` with the incorrect story... 

My question is, how can I explicitly specify the story I want to load in the `ArgsTable`?

I've tried the following so far, and none have worked:

```mdx
<ArgsTable story="storyname"/>
```

```mdx
<ArgsTable story="group-subgroup--storyname"/>
```

```mdx
<ArgsTable story="Group/Subgroup/Storyname"/>
```

Each of the above have resulted in the following being rendered:

![Screenshot_20211101_123532](https://user-images.githubusercontent.com/62898349/139707269-5767b198-0567-432c-bcb2-7b03c23dd8f1.png)

Thanks in advance for your help and guidance :smile:

**Comment by shilman** [comment]
If you're defining your stories in MDX, I believe you do the following:

```
<Story name="Foo Bar"> ... </Story>

<ArgsTable story="Foo Bar" />
```

**Reply by vinman1** [reply]
Hi @shilman , thanks for the quick response.

Actually, I am not defining my stories in MDX, I am defining them in a Typescript file using _Componrnt Story Format_, and then embedding them in a MDX documentation page using this approach:

```jsx
<Canvas>
  <Story id="group-subgroup--storyname" />
</Canvas>
```

The MDX docs page is also linked to the story with the `docs.page` parameter:

```javascript
import ComponentDocs from './my-component.docs.mdx';

...

meta.parameters = {
  docs: {
    page: ComponentDocs
  }
};
```
Essentially, I'm trying to do something like this:

```jsx
<ArgsTable story="<storyName of my embedded story>" />
```

Is it possible to feed the `id` of the story directly to the `ArgsTable` the same way I do for the `Story` component?

**Comment by Angry-Sparrow** [comment]
hope it helps.

<ArgsTable of={Your Component}/>

**Reply by vinman1** [reply]
Hi @leesgithub2019 , thanks for the suggestion.

Unfortunately, that doesn't help for the following reasons:
- I want to point my `ArgsTable` to a _specific Story_ defined for my component.
- In my specific case, I am not defining stories using the `.component` property but rather by using the `.template` one available for Angular projects.

Honestly, I think what I'd need is the ability to do the following:

```jsx
<ArgsTable id="group-subGroup--storynName">
```
where `group-subGroup--storyName` is the `id` of the story as shown in the browser's address bar and similarly used when embedding an existing story in MDX documentation file (like so):

```jsx
<Canvas>
  <Story id="group-subgroup--storyName" />
</Canvas>
```
Is this possible?

**Comment by Sebastian-G** [comment]
It seems like I have the same issue: 

I'm using latest stable `6.3.12`

```
<ArgsTable story="Design System/Components/Table" />
<ArgsTable id="design-system-components-table--basic" />
<ArgsTable of={TableComponent} />
<ArgsTable components={{ TableComponent }} />
```

results in =>

 ![ArgsTable](https://user-images.githubusercontent.com/9519879/141131991-c30eb337-a630-4a79-bc0f-5b03227ac0a0.png)


Any idea? Is it a Bug?

If I don't use the MDX to write the doc it is working 😞

**Comment by rhenness** [comment]
Also running into the same issue.

**Comment by ronald-cognite** [comment]
Same problem here.

**Comment by JoshBowdenConcepts** [comment]
Same problem as well!

**Comment by AsmisAlan** [comment]
Same issue here. storybook v6.3.0 with angular 12

**Comment by schmolie** [comment]
Same issue here. I'm running latest stable v6.4.18.

**Comment by Felipebasilio** [comment]
Same problem here!

**Reply by 9talie** [reply]
I was able to solve that problem by configuring Compodoc in my project as per indicated in this article : https://github.com/storybookjs/storybook/blob/next/addons/docs/angular/README.md

**Comment by rubanp** [comment]
I'm having the same problem.

The ArgsTable renders fine when it is generated by default but not when written with MDX.

Perhaps it's time to raise an issue regarding this?

**Reply by dmackerman** [reply]
Yeah, ArgsTable appears to just not work for many of us. I'm using React, but it never pulls the props if you do `<ArgsTable of={Component} />`. Frustrating.

**Comment by blingerson** [comment]
This is still happening, is there anyone that can provide some guidance for this? Ideally stories / docs can be written in this way because it allows for full type checking in the stories.ts file and embedding in the .mdx file. Are we just stuck with the decision of no type checking vs no working argsTable in MDX?

**Reply by blingerson** [reply]
Note, this works for PRIMARY_STORY, but not when specifying an embedded story via the story prop.

**Comment by khuezy** [comment]
@vinman1 @dmackerman @blingerson 
You need to add this:
```
export const Template = (args) => <YourComponent {...args} />

<ArgsTable of={YourComponent} />

```

You should see your component props now.

Sorry I lied, it worked for a little bit for me, now it's not showing up.

**Comment by mechadragon01** [comment]
After struggling for a bit, finally got it work with this:

`button.stories.mdx`
```javascript
<Canvas>
  <Story name='Primary' args={{ children: 'Button' }}>
    {Template.bind({})}
  </Story>
</Canvas>

<ArgsTable story='Primary' />
```

Hope this works for others using React!

**Comment by salehkaddoura-alation** [comment]
Bumping this issue as it still exists in v6.5.10

**Comment by RomainFALLER** [comment]
Still have the issue, frustrating...

**Comment by smellai** [comment]
I also have this issue, we are facing multiple issues with the storybook upgrade, we failed multiple times the tentative of moving from 6 to 7. today I tried to move to a newer 6.x version and apparently that breaks as well (as described in this ticket).
